core/app/(user)/communities/actions.ts

```diff
@@ -46,238 +47,49 @@ export const createCommunity = defineServerAction(async function createCommunity
-							.insertInto("permissions")
-							.values({
-								memberId: member?.id,
-							})
-							.returning("id")
-					)
-					.insertInto("_PermissionToStage")
-					.values((eb) => [
-						{
-							A: eb.selectFrom("new_permission").select("id"),
-							B: stages[0],
-						},
-						{
-							A: eb.selectFrom("new_permission").select("id"),
-							B: stages[1],
-						},
-						{
-							A: eb.selectFrom("new_permission").select("id"),
-							B: stages[2],
-						},
-						{
-							A: eb.selectFrom("new_permission").select("id"),
-							B: stages[3],
-						},
-					]),
-				{ communitySlug: slug }
-			).execute();
-
-			const moveConstraintPromise = autoRevalidate(
-				db.insertInto("move_constraint").values([
-					{
-						//  Submitted can be moved to: Consent, To Evaluate, Under Evaluation, Shelved
-						stageId: stages[0],
-						destinationId: stages[1],
-					},
-					{
-						stageId: stages[1],
-						destinationId: stages[2],
-					},
-					{
-						stageId: stages[2],
-						destinationId: stages[3],
-					},
-					{
-						stageId: stages[3],
-						destinationId: stages[4],
-					},
-					{
-						stageId: stages[4],
-						destinationId: stages[5],
-					},
-				]),
-				{ communitySlug: slug }
-			).execute();
-
-			const createPubPromise = autoRevalidate(
-				db
-					.with("new_pubs", (db) =>
-						db
-							.insertInto("pubs")
-							.values({
-								communityId: communityUUID,
-								pubTypeId: pubTypeId as PubTypesId,
-							})
-							.returning("id")
-					)
-					.with("pubs_in_stages", (db) =>
-						db.insertInto("PubsInStages").values((eb) => [
-							{
-								pubId: eb.selectFrom("new_pubs").select("id"),
-								stageId: stages[0],
-							},
-						])
-					)
-					.insertInto("pub_values")
-					.values((eb) => [
-						{
-							pubId: eb.selectFrom("new_pubs").select("new_pubs.id"),
-							fieldId: fields.find((field) => field.slug === `${slug}:title`)!.id,
-							value: '"The Activity of Slugs I. The Induction of Activity by Changing Temperatures"',
-						},
-						{
-							pubId: eb.selectFrom("new_pubs").select("new_pubs.id"),
-							fieldId: fields.find((field) => field.slug === `${slug}:content`)!.id,
-							value: '"LONG LIVE THE SLUGS"',
-						},
-					]),
-				{ communitySlug: slug }
-			).execute();
-			await Promise.all([permissionPromise, moveConstraintPromise, createPubPromise]);
-		}
+						.returning("community_memberships.id")
+				)
+				.insertInto("members")
+				.values((eb) => ({
+					id: eb
+						.selectFrom("community_membership")
+						.select("community_membership.id") as unknown as MembersId,
+					userId: user.id,
+					communityId: eb.selectFrom("new_community").select("new_community.id"),
+					role: MemberRole.admin,
+				}))
+				.returning("id"),
+			{ communitySlug: slug }
+		).executeTakeFirstOrThrow();
 		revalidatePath("/");
 	} catch (error) {
+		if (isUniqueConstraintError(error) && error.constraint === "communities_slug_key") {
```

Comment:
It would be nice to eventually generate these constraint names with kanel so that this line of code would cause a type error if that constraint were ever deleted or renamed. The same thing would also be useful for queries were we refer to the constraint name in an `on conflict (<constraint name>)` expression